### ISSUE TITLE
Change "back-calculation" to "nonparametric" infection model throughout

### ISSUE THREAD

**Issue by jamesmbaazam** [epiforecasts/EpiNow2#523]
At some point we should change back-calculation to nonparametric infection model throughout as that's what we're calling it in the model definition vignette.

_Originally posted by @sbfnk in https://github.com/epiforecasts/EpiNow2/pull/521#discussion_r1407455280_

**Comment by sbfnk** [comment]
To be revisited after #345

**Comment by jamesmbaazam** [comment]
To confirm, this is just a simple re-phrasing exercise right?

**Comment by seabbs** [comment]
Are we sure non-mechanistic is the right name? 

https://github.com/epiforecasts/EpiNow2/blob/dbb6536262e68b32a60dab29e3af731b11b1875e/inst/stan/functions/infections.stan#L69

The model is really a few things right: 

1. A pretty standard deconvolution approach but in the Bayesian context
2. An exponential growth model
3. A log link model (this is closest to being a "non-mechanistic model" in work with @samuelbrand we call this a direct infections model)


One option could be to break those things up either with or without an overarching naming wrapper

**Comment by seabbs** [comment]
Looking into this in the using the `estimate_infections` vignette we call this non-parametric and in the model definition vignette we call it non-mechanistic

**Comment by sbfnk** [comment]
The way I see it the key distinction from the default model is that it doesn't use the renewal equation so if we call that the mechanism (for generating new infections) it makes it non-mechanistic. We could call it "non-renewal" instead to be more explicit?

**Comment by jamesmbaazam** [comment]
> Non-renewal

Nice!

**Comment by seabbs** [comment]
One option would be a larger breaking change to make the interface: 

For renewal

```r
estimate_infections(...
  model = rt_opts()
...
)
```

For deconvolution

```r
estimate_infections(
	model = deconvolution_opt()
```

For growth rate


```r
estimate_infections(
	model = growth_opts()
)
```

**Comment by sbfnk** [comment]
I really like this idea - for names perhaps we could use

```r
estimate_infections(...
  model = renewal_opts()
...
)
```

instead of `rt_opts()`?

> For growth rate
> 
> ```r
> estimate_infections(
> 	model = growth_opts()
> )
> ```

We never implemented this as it's the same as renewal with `gt = Fixed(1)`.

**Comment by sbfnk** [comment]
Noting that the original issue was about wording in the docs so whilst obviously related the change of options might be better suited to a separate issue/PR combo.

**Comment by sbfnk** [comment]
> non-renewal

On reflection I think this is the most explicit but it's not what people will search for or be familiar with when choosing this model. That would be an argument of calling it "deconvolution".

**Comment by seabbs** [comment]
My point was realoly I would be favour of closing this as a not do an update the interface instead.

> We never implemented this as it's the same as renewal with gt = Fixed(1).

also for another issue but this would open up having syntax sugar to make this more visible.

**Comment by sbfnk** [comment]
> My point was really I would be favour of closing this as a not do 

I think we still want to have consistent wording in the doc (which is what the original issue was) - if we change the interface to use `deconvolution_opts()` and call it backcalculation in the docs I think it'll just be confusing.

**Comment by seabbs** [comment]
yes but changing the interface changes the naming as there wont be two binary categories anymore

**Comment by sbfnk** [comment]
sorry, I'm not following - isn't it still two categories of `renewal_model()` vs. `deconvolution_model()` so we should update the language in the vignettes etc. accordingly?

**Comment by sbfnk** [comment]
pushing this to the next release as it's turning into a breaking change and we already have enough things to deprecate.

**Comment by seabbs** [comment]
L> so we should update the language in the vignettes etc. accordingly?

because there is more than one model in the decovolution approach. I think this is all getting a bit lost. Essentially we need to make the docs match the code and as a follow up issue consider a different interface where we nest models less and each has its own opts function
